wolfcrypt/src/rsa.c

```diff
@@ -881,18 +886,24 @@ static int RsaMGF1(enum wc_HashType hType, byte* seed, word32 seedSz,
         /* find largest amount of memory needed which will be the max of
          * hLen and (seedSz + 4) since tmp is used to store the hash digest */
         tmpSz = ((seedSz + 4) > (word32)hLen)? seedSz + 4: (word32)hLen;
+        if (tmpSz > RSA_MAX_SIZE/8)
```

Comment:
This check is only needed for the case when XMALLOC is not used.

Reply:
Seems confusing but it's outside the XMALLOC logic.

Reply:
The check should only be done when XMALLOC is not used, since the stack buffer is a fixed size.